File: src/main/java/tech/jhipster/lite/shared/github/infrastructure/primary/RestGithubToken.java
package tech.jhipster.lite.shared.github.infrastructure.primary;

import tech.jhipster.lite.shared.github.domain.authentication.GithubToken;

public record RestGithubToken(String accessToken, String tokenType) {
  public static RestGithubToken from(GithubToken token) {
    return new RestGithubToken(token.accessToken(), token.tokenType());
  }
}

File: src/main/webapp/app/module/secondary/RestGithubToken.ts
import { GithubToken } from '@/module/domain/GithubToken';
import { AxiosResponse } from 'axios';

export interface RestGithubToken {
  accessToken: string;
  tokenType: string;
}

export const mapToGithubToken = (response: AxiosResponse<RestGithubToken>): GithubToken => {
  const data = response.data;
  return {
    accessToken: data.accessToken,
    tokenType: data.tokenType,
  };
};
